src/components/FilterPanel.jsx

```diff
@@ -15,41 +15,61 @@ import AmountFilter from './AmountFilter';
 import GradeLevelFilter from './GradeLevelFilter';
 import MajorFilter from './MajorFilter';
 import CloseIcon from '@mui/icons-material/Close';
+import WarningAmberOutlinedIcon from '@mui/icons-material/WarningAmberOutlined';
 import PropTypes from 'prop-types';
 
 export default function FilterPanel({ onClose }) {
-  const [{ minAmount, maxAmount, grades, majors }, setQueryParam] =
-    useQueryParams();
+  const [params, setQueryParams] = useQueryParams();
+
+  const [minAmount, setMinAmount] = useState(params.minAmount);
+  const [maxAmount, setMaxAmount] = useState(params.maxAmount);
+  const [grades, setGrades] = useState(params.grades);
+  const [majors, setMajors] = useState(params.majors);
 
   const filters = {
-    Major: (
-      <MajorFilter
-        majors={majors}
-        onSelect={(m) => setQueryParam('majors', m)}
-        onDelete={(m) =>
-          setQueryParam(
-            'majors',
-            majors.filter((major) => major !== m)
-          )
-        }
-      />
-    ),
-    Amount: (
-      <AmountFilter
-        min={minAmount ?? 0}
-        max={maxAmount ?? 0}
-        onMinChange={(val) => setQueryParam('minAmount', val)}
-        onMaxChange={(val) => setQueryParam('maxAmount', val)}
-      />
-    ),
-    'Grade Level': (
-      <GradeLevelFilter
-        grades={new Set(grades)}
-        changeFn={(e) => setQueryParam('grades', e)}
-      />
-    ),
+    Major: {
+      comp: (
+        <MajorFilter
+          majors={majors}
+          onSelect={(m) => setMajors(m)}
+          onDelete={(m) => setMajors(majors.filter((major) => major !== m))}
+        />
+      ),
+      wasChanged:
+        JSON.stringify(majors?.length > 0 ? majors : undefined) !==
+        JSON.stringify(params.majors),
+    },
+    Amount: {
+      comp: (
+        <AmountFilter
+          min={minAmount ?? 0}
+          max={maxAmount ?? 0}
+          onMinChange={(val) => setMinAmount(parseInt(val))}
+          onMaxChange={(val) => setMaxAmount(parseInt(val))}
+        />
+      ),
+      wasChanged:
+        (minAmount || undefined) !== params.minAmount ||
+        (maxAmount || undefined) !== params.maxAmount,
+    },
+    'Grade Level': {
+      comp: (
+        <GradeLevelFilter
+          grades={new Set(grades)}
+          changeFn={(e) => setGrades(e)}
+        />
+      ),
+      wasChanged:
+        JSON.stringify(grades?.length > 0 ? grades : undefined) !==
```

Comment:
Again I'm not understanding why this doesn't work:
```suggestion
        JSON.stringify(grades) !==
```